Frameworks/UIKit.Xaml/TextField.xaml.h

```diff
@@ -0,0 +1,268 @@
+            } else {
+                return TextBox->FocusState != Windows::UI::Xaml::FocusState::Unfocused;
+            }
+        }
+    }
+
+    property Windows::UI::Xaml::VerticalAlignment VerticalTextAlignment
+    {
+        Windows::UI::Xaml::VerticalAlignment get()
+        {
+            return (Windows::UI::Xaml::VerticalAlignment)GetValue(s_textVerticalAlignmentProperty);
+        }
+        void set(Windows::UI::Xaml::VerticalAlignment value)
+        {
+            SetValue(s_textVerticalAlignmentProperty, value);
+            _SetTextVerticalAlignment();
+        }
+    }
+
+    property UIKit::Xaml::TextBorderStyle BorderStyle
+    {
+        UIKit::Xaml::TextBorderStyle get()
+        {
+            return (UIKit::Xaml::TextBorderStyle)GetValue(s_borderStyleProperty);
+        }
+        void set(UIKit::Xaml::TextBorderStyle value)
+        {
+            SetValue(s_borderStyleProperty, value);
+            _SetBorderStyle();
+        }
+    }
+
+    void ApplyFont(Platform::String^ fontFamily, Windows::UI::Text::FontStretch strech, Windows::UI::Text::FontStyle fontStyle, double fontSize, Windows::UI::Text::FontWeight fontWeight);
+
+internal:
+    property Windows::UI::Xaml::Controls::TextBox^ TextBox {
+        Windows::UI::Xaml::Controls::TextBox^ get();
+    }
+
+    property Windows::UI::Xaml::Controls::PasswordBox^ PasswordBox {
+        Windows::UI::Xaml::Controls::PasswordBox^ get();
+    }
+
+    // Static entry point to register our dependency properties
+    static void _RegisterDependencyProperties();
+
+    // Hook pointer events
+    void RegisterEventsHandlers(
+        const Microsoft::WRL::ComPtr<ABI::Windows::UI::Xaml::IRoutedEventHandler>& textChangedHandler,
+        const Microsoft::WRL::ComPtr<ABI::Windows::UI::Xaml::IRoutedEventHandler>& gotFocusHandler,
+        const Microsoft::WRL::ComPtr<ABI::Windows::UI::Xaml::IRoutedEventHandler>& lostFocusHanlder,
+        const Microsoft::WRL::ComPtr<ABI::Windows::UI::Xaml::IRoutedEventHandler>& enterKeyDownHandler);
+
+    void UnregisterEventHandlers();
+
+    // programatically kills the focus on this control
+    void KillFocus();
+
+    // programatically set focus on this control
+    void SetFocus();
+
+    // programatically setting focus on this control
+    bool BecomeFirstResponder();
+
+private:
+    static Windows::UI::Xaml::DependencyProperty^ s_textProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_secureEntryProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_placeholderProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_inputScopeProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_enabledProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_foregroundProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_textAlignmentProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_textVerticalAlignmentProperty;
+    static Windows::UI::Xaml::DependencyProperty^ s_borderStyleProperty;
```

Comment:
text is two way binding, which i am keeping. for the rest, i changed them to properties.

---
In reply to: [113789238](https://github.com/Microsoft/WinObjC/pull/2616#discussion_r113789238) [](ancestors = 113789238,113753740,113595069,113562562)